jupyter_server_ydoc/app.py

```diff
@@ -41,14 +41,47 @@ class YDocExtension(ExtensionApp):
     )
 
     ystore_class = Type(
-        default_value=JupyterSQLiteYStore,
+        default_value=sqlite_ystore_factory(),
         klass=BaseYStore,
         config=True,
         help="""The YStore class to use for storing Y updates. Defaults to JupyterSQLiteYStore,
         which stores Y updates in a '.jupyter_ystore.db' SQLite database in the current
         directory, and clears history every 24 hours.""",
     )
 
+    sqlite_ystore_db_path = Unicode(
+        ".jupyter_ystore.db",
+        config=True,
+        help="""The path to the YStore database. Defaults to '.jupyter_ystore.db' in the current
+        directory. Only applicable if the YStore is an SQLiteYStore.""",
+    )
+
+    @observe("sqlite_ystore_db_path")
+    def _observe_sqlite_ystore_db_path(self, change):
+        if issubclass(self.ystore_class, SQLiteYStore):
+            self.ystore_class.db_path = change["new"]
+        else:
+            raise RuntimeError(
+                f"ystore_class must be an SQLiteYStore to be able to set sqlite_ystore_db_path, not {self.ystore_class}"
+            )
+
+    sqlite_ystore_document_ttl = Int(
+        None,
+        allow_none=True,
+        config=True,
+        help="""The document time-to-live in seconds. Defaults to None (document history is never
+        cleared). Only applicable if the YStore is an SQLiteYStore.""",
+    )
+
+    @observe("sqlite_ystore_document_ttl")
```

Comment:
Could you elaborate on the difference between using the `validate` and `observe` traitlets decorators?

Reply:
Yes I was also hesitating, I think there is some overlap, but here `sqlite_ystore_document_ttl` is only used to set the corresponding attribute in the YStore class, so it seems there's nothing to validate since nobody else consumes it.